datafusion/common/src/scalar.rs

```diff
@@ -1987,3 +1987,1176 @@ impl ScalarType<i64> for TimestampNanosecondType {
         ScalarValue::TimestampNanosecond(r, None)
     }
 }
+
```

Comment:
these tests are moved, verbatim. The only change is that this line

```rust
    use arrow::{array::*, datatypes::*};
```

was removed at the direction of the rust compiler as it was no longer used